### DISCUSSION TITLE
Could the battery charge limiter work for other notebooks?

### DISCUSSION THREAD

**Discussion by LoreAPC** [seerge/g-helper#306] [General]
Could the endpoint discovery work for other notebooks as well?
I've got a Huawei Matebook and the only reason to keep their bloated software installed is to limit the battery charge.

How did you discover the correct command to limit battery charge on the Zephyrus?

Could a similar command work for other brands?

**Comment by seerge** [comment]
@LoreAPC it's usually brand / laptop specific :) unless your model supports that by design, it can't be added

**Comment by LoreAPC** [comment]
The function is absolutely there because "Huawei PC Manager" can limit the battery charge.
Question is how can I try to find the correct command?

**Reply by seerge** [reply]
Then i would look for open source apps for huawei, and see if they can do it and how they do it :) I absolutely can't help here, as i don't own device and have no idea how things work there ;) usually it's some low level WMI methods that can set this setting in bios that do that

**Comment by steverand** [comment]
The charge limit does not work on my ASUS A17 FA706IC but changing the registry key here does:

Computer\HKEY_LOCAL_MACHINE\SOFTWARE\ASUS\ASUS System Control Interface\AsusOptimization\ASUS Keyboard Hotkeys

ChargingRate - decimal 80 = 80% charge limit.

Changing in GHelper does not change this value for me.

**Reply by steverand** [reply]
After stopping at 80% I changed GHelper to 85 to see if it would have any effect and it then charged to 100. =-o
Now my registry says 100. I suspect changing the slider at all resets it to 100 so i'll try again at 80 in the registry without touching GHelper.

**Reply by seerge** [reply]
G-helper doesn't touch registry. What you explain is asus optimization service interfering, and setting it's own limits. That's why it didn't work fromg-helper
